src/CMakeLists.txt

```diff
@@ -17,9 +19,48 @@ add_library(settings
             setting_type_int.c
             setting_type_str.c)
 
-target_link_libraries(settings)
-
 target_include_directories(settings PUBLIC ${PROJECT_SOURCE_DIR}/include)
+set_property(TARGET settings PROPERTY C_STANDARD 99)
+set_property(TARGET settings PROPERTY C_STANDARD_REQUIRED ON)
+set_property(TARGET settings PROPERTY C_EXTENSIONS OFF)
+
+if(EXISTS ${LIBSBP_PATH}/c/include/)
+    target_include_directories(settings PUBLIC ${LIBSBP_PATH}/c/include/)
+else()
+    message(WARNING "No LibSbp include path found.")
+endif()
+
+if (EXISTS ${LIBSWIFTNAV_PATH}/include/)
+    target_include_directories(settings PRIVATE ${LIBSWIFTNAV_PATH}/include)
+else()
+    message(WARNING "No LibSwiftNav include path found.")
+endif()
+
+if (NOT MSVC)
```

Comment:
Maybe we can do something like
```
if (MSVC)
    target_compile_options(settings PRIVATE /Wall)
else ()
    target_compile_options(settings PRIVATE
            -Wall -Wextra -Wno-strict-prototypes -Werror -fno-unwind-tables
            -fno-asynchronous-unwind-tables -Wimplicit -Wshadow -Wswitch-default
            -Wswitch-enum -Wundef -Wuninitialized -Wcast-align -Wformat=2
            -Wimplicit-function-declaration -Wredundant-decls -Wformat-security
            -ggdb)
endif ()
```

`/WX` can't be added at this point as there are plenty of warnings in the msvc build, I have a separate issue for fixing those.

Reply:
Sure thing!